src/Hydrator/ModelHydrator.php

```diff
@@ -13,30 +13,23 @@
  */
 final class ModelHydrator implements Hydrator
 {
+    private $arrayHydrator;
+    private $creatableFromArray;
+
+    public function __construct(ArrayHydrator $arrayHydrator, CreatableFromArray $creatableFromArray)
```

Comment:
Of course you do. Check my PR. Where is the hydrator instanciated? Which classes implement CreatableFromArray? 

With this, you cannot reuse the same hydrator for two endpoints.